spec.bs

```diff
@@ -287,18 +287,383 @@ href="https://html.spec.whatwg.org/multipage/embedded-content-other.html#dimensi
+of their fenced frame configs in order to ensure that they have considered the privacy implications
+of each field, though they may choose to set the values to null.
+
+Each time a <{fencedframe}> navigates to a [=fenced frame config=], it is instantiated as a new
+[=fenced frame config instance=], which governs that particular context inside the [=fenced
+navigable container/fenced navigable=].
+
+<h4 id=fenced-frame-config-struct>The [=fenced frame config=] [=struct=]</h4>
+
+We now establish some preliminary types:
+
+A <dfn export for=fencedframeconfig>visibility</dfn> is either "<dfn export for=visibility>
+`opaque`</dfn>" or "<dfn export for=visibility>`transparent`</dfn>".
+
+A <dfn export for=fencedframetype>mapped url</dfn> is a [=url=]. <span class=XXX>TODO: Stipulate a
+url with particular schemes?</span>
+
+A <dfn export for=fencedframetype>size</dfn> is a struct with non-negative integer <dfn export for=
+"size">width</dfn> and non-negative integer <dfn export for="size">height</dfn>. <span class=XXX>
+TODO: Maybe change the numeric type.</span>
+
+An <dfn export for=fencedframetype>interest group descriptor</dfn> is a struct with <dfn export
+for="interest group descriptor">owner</dfn>, which is a string, and <dfn export for="interest group
+descriptor">name</dfn>, which is a string.
+
+An <dfn export for=fencedframetype>exhaustive set of sandbox flags</dfn> is a [=sandboxing flag
+set=].
+
+An <dfn export for=fencedframetype>exhaustive set of permissions</dfn> is a [=list=] of
+[=policy-controlled features=].
+
+A <dfn export for=fencedframetype>fenced frame reporter</dfn> is <span class=XXX>TODO: Specify the type for this.</span>
+
+<div algorithm>
+  In order to <dfn export>report an event</dfn>, run these steps:
+
+  1. <span class=XXX>TODO: fill this in</span>
+</div>
+
+<div algorithm>
+  In order to <dfn export>report a private aggregation event</dfn>, run these steps:
+
+  1. <span class=XXX>TODO: Fill this in</span>
+</div>
+
+<div algorithm>
+  In order to <dfn export>set automatic beacon data</dfn>, run these steps:
+
+  1. <span class=XXX>TODO: Fill this in</span>
+</div>
+
+An <dfn export for=fencedframetype>exfiltration budget metadata</dfn> is a struct containing an
+<dfn export for="exfiltration budget metadata">origin</dfn>, which is an [=origin=]; and an <dfn
+export for="exfiltration budget metadata">amount to debit</dfn>, which is a non-negative valid
+floating point number.
+
+An <dfn export for=fencedframetype>exfiltration budget metadata reference</dfn> is a struct
+containing an <dfn export for="exfiltration budget metadata reference">origin</dfn>, which is an
+[=origin=]; and an <dfn export for="exfiltration budget metadata reference">amount to debit
+reference</dfn>, which is a mutable reference to a non-negative valid floating point number.
+<span class=XXX>TODO: are mutable references a thing in spec?</span>
```

Comment:
Can this just be the number itself? Specs have the concept of a "pointer to foo" basically, but if we could make this just a plain ole number (instead of a pointer to a number elsewhere) that would be easiest. Is there a strong reason not to do this?

Reply:
It has to be a reference because when you perform a fence-escaping navigation, it subtracts this number from the budget and then sets it to 0 (through the reference, so that it applies to all instances created from the same config).

Reply:
Where does "the budget" that we subtract _from_ ultimately live? Right now in our spec it seems like a closed loop where the config member is referenced by the instance, and the instance member is referenced by the instantiation algorithm.